Add section framework

Azmaeus opened the pull request:
This pulls the Header out to its own individual file under the newly-created components directory.

BEFORE:
![image](https://user-images.githubusercontent.com/29641878/223274017-42a4e277-cce9-4e0c-8cdb-13dd0be0546d.png)

AFTER:
![image](https://user-images.githubusercontent.com/29641878/223588623-96ec22a1-b868-4d1e-a6d5-61e980cf81e7.png)

Azmaeus committed:
Move background and font color from Header to App

Azmaeus committed:
Create section class

Azmaeus committed:
Break-out App JSX into separate functions

bobjohnbob commented on src/App.js:
```diff
@@ -4,15 +4,57 @@ import './App.css';
+  );
+}
+
+function Header() {
```
I would move each of these to their own files.  If you create a Header.js  file and `export default Header` at the bottom, you can import it at the top of this file like this `import Header from './Header';` and then use it like you're using it here.

Azmaeus replied:
Thanks for the suggestion!

bobjohnbob commented on src/App.js:
```diff
@@ -4,15 +4,57 @@ import './App.css';
+  );
+}
+
+function Story() {
```
This can be another file as well.  A function that can be used this way in React is typically referred to as a "component".

jesseolsen replied:
Components are often all put under a "components" directory (but not required).

Azmaeus replied:
Thanks for the suggestion! I've updated this branch to now have components in a "components" directory and in their own files.

bobjohnbob commented on src/App.js:
```diff
@@ -4,15 +4,57 @@ import './App.css';
+
+function Story() {
+  return (
+    <div className="section">
```
You can actually use a `<section></section>` element for this https://developer.mozilla.org/en-US/docs/Web/HTML/Element/section

jesseolsen replied:
If you really want to make your page accessible for the blind, etc., you can Google "Axe Dev Tools chrome extention" and install that, then run the scanner on your page.  If you want to install it, I can show you how it works, but it's pretty easy.  Again, optional, but at Procore I try to get everyone to think about accessibility while they are developing their code.

Azmaeus replied:
Thanks @bobjohnbob I'll look into the section element and see if it fits what I'm planning here. This "section" class will be used for smoothscroll later to indicate what to scroll to.

@jesseolsen I do want to keep accessibility in mind, thanks for the tip!

bobjohnbob commented on src/App.js:
```diff
@@ -4,15 +4,57 @@ import './App.css';
-        </h3>
-      </header>
+      <Header />
+      {/* <Story />
```
Rather than commenting this out.  Maybe create a new branch off of master to be a staging branch.  Then we can target staging with all the things aren't ready to go live.  Then when you're ready to go live with a set of changes, merge staging into master.

Azmaeus replied:
Sounds good. I'll work on changing my workflow to use a staging step soon. For now, I broke each component into their own branch and removed all but Header from this PR.

bobjohnbob reviewed (changes requested):
See comments.  I'm out of the habit of leaving them in an actual review. I usually just leave them separate from the actual review but you should be able to see them all.

jesseolsen commented on src/App.js:
```diff
@@ -4,15 +4,57 @@ import './App.css';
+function Header() {
+  return (
+    <header className="App-header">
+      <img src={logo} className="App-logo" alt="logo" />
```
Looks like all references to "logo" should now be to your "profile photo"

Azmaeus replied:
Yeah, I reused logo for simplicity, but I agree it should reflect what it really is. I've implemented this change, now it's either profile-photo (general), profile_photo (on import where hyphen isn't allowed), or "profile photo" (on img alt for accessibility).

jesseolsen commented on src/App.js:
```diff
@@ -4,15 +4,57 @@ import './App.css';
+  return (
+    <header className="App-header">
+      <img src={logo} className="App-logo" alt="logo" />
+      <h1>
+        Eric F. Olsen
+      </h1>
```
You could put `<h1>...</n1>` and the other headers on 1 line

Azmaeus replied:
I went ahead and put <h1>Eric F. Olsen</h1> on one line, and <h3>Software Engineer</h3> on a separate line, I feel that is more readable. Is there a reason to put them both on the same line?

Azmaeus replied:
Based on your other comment, I've scrapped the h3 and replaced it with p and a css class to adjust it.

jesseolsen commented on src/App.js:
```diff
@@ -4,15 +4,57 @@ import './App.css';
+      <h1>
+        Eric F. Olsen
+      </h1>
+      <h3>
+        Software Engineer
+      </h3>
```
For accessibility, you are not supposed to skip heading levels.  In this case, it doesn't look like it's actually a header, so you could just make it a <div> and pass in attributes to make it look how you want it to look.  If you want to get into a 3rd party UI, with fanciar buttons than the vanilla buttons and text, etc., you may want to look into using Material UI (MUI).

Azmaeus replied:
@jesseolsen That's good to know! I'm reading-up on that now, but if you have a minute the HTML was taken in your comment so it didn't display properly here, would you please retry with it in a code block?

Azmaeus replied:
@jesseolsen from what I've read, I'm guessing you suggested to use `<p class="subtitle">Software Engineer</p>` or something to that effect, then style it to be the size I want?

Azmaeus replied:
I made the change from h3 to p with a css class. Please see the updated BEFORE and AFTER screenshots. It does change how "Software Engineer" looks, but I like it not bold, and played with the size to where I think it looks good. Thoughts?

Azmaeus committed:
Set vscode default tab size

The auto-detect was setting new files to 4 spaces, but the create-react-app boilerplate was using 2 spaces.

Azmaeus committed:
Move header into its own file, rename headshot to profile photo

Azmaeus committed:
Remove underscore from img alt

Azmaeus committed:
Update header to avoid skipping heading levels

For accessibility, avoid using heading elements (e.g. h1, h2, etc) for anything other than headings (new sections) because assistive technology users might think they missed something.

Azmaeus commented:
I've made a lot of changes based on the recommendations provided. Please re-review the code, and also the BEFORE and AFTER pictures that previously showed what was commented-out (and is now removed), and now shows the new style of the occupation name.

bobjohnbob reviewed (approved):
:+1:
